src/probnum/testproblems/linalg/_random_spd_matrix.py

```diff
@@ -0,0 +1,158 @@
+"""Random symmetric positive definite matrices."""
+
+from typing import Optional, Sequence
+
+import numpy as np
+import scipy.stats
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, RandomStateArgType
+
+
+def random_spd_matrix(
+    dim: IntArgType,
+    spectrum: Sequence = None,
+    random_state: Optional[RandomStateArgType] = None,
+) -> np.ndarray:
+    """Random symmetric positive definite matrix.
+
+    Constructs a random symmetric positive definite matrix from a given spectrum. An
+    orthogonal matrix :math:`Q` with :math:`\\operatorname{det}(Q)` (a rotation) is
+    sampled with respect to the Haar measure and the spectrum is rotated accordingly
+    resulting in :math:`A=Q \\operatorname{diag}(\\lambda_1, \\dots, \\lambda_n)
+    Q^\\top`. If no spectrum is provided, one is randomly drawn from a Gamma
+    distribution.
```

Comment:
Holy moly it's good that we store those functions somewhere now, given what happens in the random spd functions :D